nni/compression/pytorch/quantization_speedup/integrated_tensorrt.py

```diff
@@ -0,0 +1,277 @@
+            tensorrt/api/python_api/infer/Int8/Calibrator.html for detail
+        calibrate_data : numpy array
+            The data using to calibrate quantization model
+        calibration_cache : str
+            The path user want to store calibrate cache file
+        batchsize : int
+            The batch size of calibration and inference
+        input_names : list
+            Input name of onnx model providing for torch.onnx.export to generate onnx model
+        output_name : list
+            Output name of onnx model providing for torch.onnx.export to generate onnx model
+        """
+        super().__init__(model, config)
+        self.model = model
+        self.onnx_path = onnx_path
+        self.input_shape = input_shape
+        self.config = config
+        self.extra_layer_bit = extra_layer_bit
+        self.strict_datatype = strict_datatype
+        self.calibrate_type = calibrate_type
+        self.calib_data_loader = calib_data_loader
+        self.calibration_cache = calibration_cache
+        self.batchsize = batchsize
+        self.input_names = input_names
+        self.output_names = output_names
+        self.context = None
+        self.onnx_config = {}
+
+    def compress(self):
+        """
+        Get onnx config and build tensorrt engine.
+        """
+        assert self.model is not None
+        assert self.onnx_path is not None
+        assert self.input_shape is not None
+
+        # Convert pytorch model to onnx model and save onnx model in onnx_path
+        _, self.onnx_config = fonnx.torch_to_onnx(self.model, self.config, input_shape=self.input_shape,
+            model_path=self.onnx_path, input_names=self.input_names, output_names=self.output_names)
+
+        if self.calib_data_loader is not None:
+            assert self.calibrate_type is not None
+            context = self.tensorrt_build_withcalib(self.onnx_path)
+        else:
+            context = self.tensorrt_build_withoutcalib(self.onnx_path)
+        self.context = context
+
+    def tensorrt_build_withcalib(self, onnx_path):
+        # convert pytorch tensor to numpy darray
+        calib_data_set = []
+        for data, _ in self.calib_data_loader:
+            calib_data_set.append(data)
+        calib_data = np.concatenate(calib_data_set)
+        calib = calibrator.Calibrator(calib_data, self.calibration_cache, self.batchsize, self.calibrate_type)
+
+        # build inference engine with calibration
+        engine = build_engine(onnx_path, calib, self.onnx_config, self.extra_layer_bit, self.strict_datatype)
+        return engine.create_execution_context()
+
+    def tensorrt_build_withoutcalib(self, onnx_path):
+        # build inference engine without calibration
+        engine = build_engine_without_calib(onnx_path, self.onnx_config, self.extra_layer_bit, self.strict_datatype)
+        return engine.create_execution_context()
+
+    def inference(self, test_data):
```

Comment:
so tensorrt does not support exporting compressed model?

Reply:
It supports exporting. Have added related function `export_quantized_model()` and `load_quantized_model()`